aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AppSyncCustomNetworkInvoker.java

```diff
@@ -156,12 +161,36 @@ public String region() {
                                                    return persistentOfflineMutationObject.region;
                                                }
                                            });
-                                       } catch (Exception e) {
-                                           persistentMutationsCallback.onFailure(new PersistentMutationsError(persistentOfflineMutationObject.responseClassName,
-                                                   persistentOfflineMutationObject.recordIdentifier, new ApolloNetworkException("S3 upload failed.", e)));
+                                       }
+                                       catch (AmazonClientException e) {
+                                           if (e.isRetryable()) {
```

Comment:
Can we investigate if we can find underlying cause? like IOException?

Reply:
Good call. Checking specifically for IOException and only retrying in that case.